### PR TITLE
ENT-4242-postgres11-migration-improvements PR

### PR DESCRIPTION
**Pull request by Lex-2008**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a pull request! Maybe @craigcomstock can review this?

**Comment by Lex-2008** [comment]
@cf-bottom build only one hub, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1703/)

https://ci.cfengine.com/job/pr-pipeline/1703/

File: packaging/common/cfengine-hub/preinstall.sh [hunk 1]
```diff
@@ -104,7 +104,28 @@ if is_upgrade && egrep '^3\.([6-9]|1[01])\.' "$PREFIX/UPGRADED_FROM.txt" >/dev/n
...
+  else
+    # Copy creating hardlinks failed, so remove partially-copied data and try simple copying
+    rm -rf "$BACKUP_DIR/data"
+    if cp -a "$PREFIX/state/pg/data" "$BACKUP_DIR"; then
+      # Copy succeeded - so we can delete old dir.
+      rm -rf "$PREFIX/state/pg/data"
+    else
+      # Copy failed, so remove partially-copied data and abort
+      rm -rf "$BACKUP_DIR/data"
+      cf_console echo "Creating of backup failed"
```
Comment (vpodzime) [inline packaging/common/cfengine-hub/preinstall.sh]: *Backup creation failed*

**Comment by Lex-2008** [comment]
this concerns hub scripts, and luckily we don't run them on exotics :-)

**Pull request merged**